src/parser/jobs/mnk/BuffUptime.js

```diff
@@ -0,0 +1,172 @@
+			if (this._lastDragonKickUse[event.targetID]) {
+				this._gcdsSinceDK[event.targetID] += 1
+			}
+
+			if (this._lastTwinSnakesUse !== null) {
+				this._gcdsSinceTS++
+			}
+		}
+	}
+
+	_onDragonKick(event) {
+		// Make sure we're tracking for this target
+		const lastApplication = this._lastDragonKickUse[event.targetID]
+
+		// If it's not been applied yet set it and skip out
+		if (!lastApplication) {
+			this._lastDragonKickUse[event.targetID] = event.timestamp
+			return
+		}
+
+		// This logic is busted if the player's GCD is under 1.67s
+		if (this._gcdsSinceDK[event.targetID] < GCD_CYCLE_LENGTH) {
+			this._earlyDragonKicks++
+		}
+
+		this._lastDragonKickUse[event.targetID] = event.timestamp
+		this._gcdsSinceDK[event.targetID] = 0
+	}
+
+	_onTwinSnakes(event) {
+		// If it's not been applied yet set it and skip out
+		if (!this._lastTwinSnakesUse) {
+			this._lastTwinSnakesUse = event.timestamp
+			return
+		}
+
+		if (this._gcdsSinceTS < GCD_CYCLE_LENGTH) {
+			this._earlyTwinSnakes++
+		}
+
+		this._lastTwinSnakesUse = event.timestamp
+		this._gcdsSinceTS = 0
+	}
+
+	_onComplete() {
+		this.checklist.add(new Rule({
+			name: 'Keep Dragon Kick up',
+			description: <Fragment>
+				Dragon Kick's blunt resistance debuff should always be applied to your primary target.
+			</Fragment>,
+			requirements: [
+				new Requirement({
+					name: <Fragment><ActionLink {...ACTIONS.DRAGON_KICK} /> uptime</Fragment>,
+					percent: () => this.getDebuffUptimePercent(STATUSES.BLUNT_RESISTANCE_DOWN.id),
+				}),
+			],
+		}))
+
+		this.checklist.add(new Rule({
+			name: 'Keep Twin Snakes up',
+			description: <Fragment>
+				Twin Snakes is an easy 10% buff to your DPS across the board.
+			</Fragment>,
+			requirements: [
+				new Requirement({
+					name: <Fragment><ActionLink {...ACTIONS.TWIN_SNAKES} /> uptime</Fragment>,
+					percent: () => this.getBuffUptimePercent(STATUSES.TWIN_SNAKES.id),
+				}),
+			],
+		}))
+
+		if (this._earlyDragonKicks) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.DRAGON_KICK.icon,
+				content: <Fragment>
+					Avoid refreshing <ActionLink {...ACTIONS.DRAGON_KICK} /> signficantly before its expiration as you're trading it for the higher damage <ActionLink {...ACTIONS.BOOTSHINE} />.
+				</Fragment>,
+				severity: SEVERITY.MEDIUM,
+				why: <Fragment>
+					{this._earlyDragonKicks} rotation cycles were interrupted by early refreshes.
+				</Fragment>,
+			}))
+		}
+
+		if (this._earlyTwinSnakes) {
+			const lostTruePotency = this._earlyTwinSnakes * (ACTIONS.TRUE_STRIKE.potency - ACTIONS.TWIN_SNAKES.potency)
+
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.TWIN_SNAKES.icon,
+				content: <Fragment>
+					Avoid refreshing <ActionLink {...ACTIONS.TWIN_SNAKES} /> signficantly before its expiration as you're losing uses of the higher potency <ActionLink {...ACTIONS.TRUE_STRIKE} />.
+				</Fragment>,
+				severity: SEVERITY.MAJOR,
```

Comment:
Do you really want to shout a MAJOR at people who make the mistake once over an entire fight?